be/Controller/UsersCtrl.php

```diff
@@ -28,10 +27,85 @@ public function __construct(
 
     public function getUsers(Response $response): Response
     {
-        $usersEntities = $this->userRepository->findBy(['deleted' => 0],
-            ['rating' => 'DESC', 'code' => 'ASC']);
-        $respArray = Helpers::entitiesListToArray($usersEntities);
-        return $response->withJson($respArray);
+        $userEntityList = $this->userRepository->findBy(
+            ['deleted' => 0],
+            ['rating' => 'DESC', 'code' => 'ASC']
+        );
+
+        $userArrayList = array_map(function (User $user) {
```

Comment:
Doszło dużo logiki do tej klasy. Teraz ta klasa zaujme się pobieraniem userów i updatem userów. Chyba powinien wydzilić pobierainie do osobnej klasy. Ale właśnie też się zastanawiam jak to jest z tymi kontrolerami, bo teoretycznie kontrolery zajmują się odebranie requesta i zrobieniem responsa. Czy w klasach kontrolerów powinna być jakaś logika na bazie czy też zawsze logikę na bazie powinno się wydzielać poza klasy kontrolerów ?

Reply:
Można by rzec, że kontrolery powinny mieć jak najmniej logiki, i jedynie korzystać z dostępnych części aplikacji, a następnie składać je w całość. W Leonie robi się czasami to właśnie w sposób poprawny. W Symfony taką logikę przechowują właśnie tak zwane "serwisy", które mają swój osobny service container. One są tam wrzucane do konkretnej akcji w kontrolerze jako parametry, i wywoływana jest odpowiednia metoda publiczna.

Jednakże, ciężko mi się już wypowiedzieć np. na temat gdzie ma być robiony "flush" z Doctrine. Niby w przykładach również jest z poziomu kontrolera, ale to wtedy cały obiekt `EntityManager` trzeba importować.

Nawiązując jeszcze do:

> Chyba powinien wydzilić pobierainie do osobnej klasy.

Proponuję z początku wydzielić jak najwięcej logiki do serwisów w jakiejś przestrzeni nazw `\Services`, i wtedy zobaczyć jak duży będzie kontroler. To ogólny kontroler, więc w sumie powinien mieć możliwość robienia wielu akcji.

Reply:
@ktrzos 
założony issue